app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,182 @@
+package com.example.userinfo
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import java.util.regex.Pattern
+
+class MainActivity : AppCompatActivity() {
+    val pincodePattern = Pattern.compile("^[1-9][0-9]{5}$")
+    val emailPattern=Pattern.compile("^[A-za-z0-9_.-]+@[a-z]+\\.+(com|co.in)")
+    val userNamePattern=Pattern.compile("^[A-za-z0-9 ]*")
+    val phoneNumberPattern=Pattern.compile("^[1-9][0-9]{9}")
```

Comment:
We can move all the validation logic to another class. So that it will become re-usable